protoc-gen-grpc-gateway/httprule/compile.go

```diff
@@ -105,10 +107,11 @@ func (t template) Compile() Template {
 		}
 	}
 	return Template{
-		Version: opcodeVersion,
-		OpCodes: ops,
-		Pool:    pool,
-		Verb:    t.verb,
-		Fields:  fields,
+		Version:  opcodeVersion,
+		OpCodes:  ops,
+		Pool:     pool,
+		Verb:     t.verb,
+		Fields:   fields,
+		Template: t.template,
```

Comment:
Strictly speaking, the template format of `google.api.HttpRule` and the one of Swagger are different.   But your approach is good enough as a first step.

IIUC, `/path/{param=foo/*}/` in `google.api.HttpRule` needs to be `/path/foo/{param}` in Swagger.